### DISCUSSION TITLE
backup between bellows and znp

### DISCUSSION THREAD

**Discussion by pipiche38** [zigpy/zigpy#1016] [Q&A] [answered]
I have a question, coming from the Domoticz user community on the possibility to backup on bellows and restore on znp ?

From what I see in the bellows backup is that the file produced is totally different than the ZNP one. I understood that we get only a kind of hash code on bellows, where on ZNP we have all nwkid, ieee and important information.

**Comment by Hedda** [comment, marked as answer]
It is supported and vice versa, the files produced by zigpy-cli should not be different so maybe a bug or user error? Both use same format: 

https://github.com/zigpy/open-coordinator-backup

however I do not think wiki and readme has been updated how-to use zigpy-cli for backup and restore since the new radio API?

https://github.com/zigpy/zigpy/wiki/Coordinator-Backup-and-Migration

https://github.com/zigpy/zigpy-cli/blob/dev/README.md

https://github.com/zigpy/zigpy-cli/pull/2

https://github.com/zigpy/zigpy-cli/pull/13

https://github.com/zigpy/zigpy/pull/1006

**Reply by pipiche38** [reply]
on bellows what is backed up seems to be only a hash strings, while on zigpy you get the all information.
So could you confirm that it works because tested ?

I understood @puddly did znp to deConz but I never see anything about bellows to znp or deConz

**Reply by puddly** [reply]
What backup interface are you using?

**Reply by pipiche38** [reply]
See my comment here https://github.com/zigpy/bellows/issues/458#issuecomment-1139462695 where by using the bellows backup command the backup file is ridiculous small

I'm not sure what MattWestb was exactly meaning

**Reply by Hedda** [reply]
> So could you confirm that it works because tested ?

Yes, backup image from bellows restored to zigpy-znp and vice versa tested by many so there have even been popular videos on YouTube. 

It is quite common to see people posting in Home Assistant community forum about backing up older TI CC2530/CC2531 and restore to newer Silabs EFR32MG12/EFR32MG13/EFR32MG21, or other people backup older Silabs ETRX35x and restoring to newer TI CC2652/CC1352

**Reply by puddly** [reply]
That's the bellows command line utility. Use zigpy-cli or the backup interface and write the write-once coordinator EUI64, I've done it many times.

**Reply by pipiche38** [reply]
Thanks indeed much aligned with what is given by ZNP.

**Reply by Hedda** [reply]
FYI, puddly has now submitted patches for ZHA to Home Assistant core and frontend/GUI for backup + restore features/settings, see:

https://github.com/home-assistant/core/pull/75791

and

https://github.com/home-assistant/frontend/pull/13295

**Reply by Hedda** [reply]
> write the write-once coordinator EUI64

Note and be warned that Silicon Labs EFR32 apparently only supports write-once for coordinator EUI64.
